packages/core/src/external/athenahealth/index.ts

```diff
@@ -397,6 +437,238 @@ class AthenaHealthApi {
+  }): Promise<VitalsCreateResponse[]> {
+    const { log, debug } = out(
+      `AthenaHealth create vitals - cxId ${cxId} practiceId ${this.practiceId} patientId ${patientId} departmentId ${departmentId}`
+    );
+    const chartVitalsUrl = `/chart/${this.stripPatientId(patientId)}/vitals`;
+    try {
+      const observation = vitals.mostRecentObservation;
+      const code = this.getObservationCode(observation);
+      if (!code) {
+        throw new MetriportError("No code found for observation", undefined, {
+          cxId,
+          practiceId: this.practiceId,
+          patientId,
+          departmentId,
+          observationId: observation.id,
+        });
+      }
+      const clinicalElementId = vitalSignCodesMapAthena.get(code);
+      if (!clinicalElementId) {
+        throw new MetriportError("No clinical element id found for observation", undefined, {
+          cxId,
+          practiceId: this.practiceId,
+          patientId,
+          departmentId,
+          observationId: observation.id,
+        });
+      }
+      if (!vitals.sortedPoints || vitals.sortedPoints.length === 0) {
+        throw new MetriportError("No points found for vitals", undefined, {
+          cxId,
+          practiceId: this.practiceId,
+          patientId,
+          departmentId,
+          observationId: observation.id,
+        });
+      }
+      const payloads = vitals.sortedPoints.map(v => {
+        const vitalsData = this.createVitalsData(v, clinicalElementId);
+        if (uniqBy(vitalsData, "readingtaken").length !== vitalsData.length) {
+          throw new MetriportError("Duplicate reading taken for vitals", undefined, {
+            cxId,
+            practiceId: this.practiceId,
+            patientId,
+            departmentId,
+            observationId: observation.id,
+          });
+        }
+        return {
+          departmentid: this.stripDepartmentId(departmentId),
+          returnvitalsid: true,
+          source: "DEVICEGENERATED",
+          vitals: [vitalsData],
+          THIRDPARTYUSERNAME: undefined,
+          PATIENTFACINGCALL: undefined,
+        };
+      });
+      const settledResponses = await Promise.allSettled(
+        payloads.map(data =>
+          this.axiosInstanceProprietary.post(chartVitalsUrl, this.createDataParams(data))
+        )
+      );
+      const errors = settledResponses.flatMap(r => (r.status === "rejected" ? [r.reason] : []));
+      if (errors.length > 0) {
+        const msg = `Failure while creating some vitals @ AthenaHealth`;
+        log(`${msg}. Cause: ${errors.map(error => errorToString(error)).join(", ")}`);
+        capture.error(msg, {
+          extra: {
+            url: chartVitalsUrl,
+            cxId,
+            practiceId: this.practiceId,
+            patientId,
+            departmentId,
+            observationId: observation.id,
+            context: "athenahealth.create-vitals",
+            errors,
+          },
+        });
+      }
```

Comment:
Its odd that if we get errors from running the promises (like a network error, for example) we don't fail the command, but if we get an error from Athena (like no response body) we do.

Reply:
Fixed.

Reply:
I see you changed the code, but the original comment still holds, as in errors from settled promises don't throw while others do.

Not saying it has to, but just thought it was odd and you seemed to agree, so just checking.

Reply:
No longer relevant.